spec/acceptance/sequence_spec.rb

```diff
@@ -1,72 +1,166 @@
+
+        factory :user, aliases: [:author, :commenter] do
+          sequence(:size, "x", aliases: [:count, :length]) { |n| "user-called-#{n}" }
+        end
+      end
 
-    expect(first_value).to eq "called-1"
-    expect(second_value).to eq "called-2"
-    expect(third_value).to eq "called-3"
-  end
+      expect(generate(:size)).to eq "global-called-a"
+      expect(generate(:count)).to eq "global-called-b"
+      expect(generate(:length)).to eq "global-called-c"
 
-  it "generates aliases for the sequence that reference the same block and retains value" do
-    FactoryBot.define do
-      sequence(:size, "a", aliases: [:count, :length]) { |n| "called-#{n}" }
+      expect(generate(:user, :size)).to eq "user-called-x"
+      expect(generate(:author, :count)).to eq "user-called-y"
+      expect(generate(:commenter, :length)).to eq "user-called-z"
     end
 
-    first_value = generate(:size)
-    second_value = generate(:count)
-    third_value = generate(:length)
+    it "generates few values of the sequence" do
+      define_class("User") { attr_accessor :email }
 
-    expect(first_value).to eq "called-a"
-    expect(second_value).to eq "called-b"
-    expect(third_value).to eq "called-c"
-  end
+      FactoryBot.define do
+        sequence(:email) { |n| "global-#{n}@example.com" }
 
-  it "generates few values of the sequence" do
-    FactoryBot.define do
-      sequence :email do |n|
-        "somebody#{n}@example.com"
+        factory :user do
+          sequence(:email) { |n| "user-#{n}@example.com" }
+        end
       end
+
+      global_values = generate_list(:email, 3)
+      expect(global_values[0]).to eq "global-1@example.com"
+      expect(global_values[1]).to eq "global-2@example.com"
+      expect(global_values[2]).to eq "global-3@example.com"
+
+      user_values = generate_list(:user, :email, 3)
+      expect(user_values[0]).to eq "user-1@example.com"
+      expect(user_values[1]).to eq "user-2@example.com"
+      expect(user_values[2]).to eq "user-3@example.com"
     end
 
-    values = generate_list(:email, 2)
+    it "generates few values of the sequence with a given scope" do
+      define_class("User") { attr_accessor :name, :email }
 
-    expect(values.first).to eq("somebody1@example.com")
-    expect(values.second).to eq("somebody2@example.com")
-  end
+      FactoryBot.define do
+        factory :user do
+          sequence(:email) { |n| "#{name}-#{n}@example.com" }
+        end
+      end
+
+      test_scope = OpenStruct.new(name: "Jester")
+      user_values = generate_list(:user, :email, 3, scope: test_scope)
+
+      expect(user_values[0]).to eq "Jester-1@example.com"
+      expect(user_values[1]).to eq "Jester-2@example.com"
+      expect(user_values[2]).to eq "Jester-3@example.com"
+    end
+  end # "on success"
+
+  # = On Failure
+  # ======================================================================
+  #
+  describe "on failure" do
+    it "it fails with an unknown sequence or factory name" do
```

Comment:
There are some `it "it` that could be quickly fixed with a search and replace, hehe